rust/lance/src/io/commit.rs

```diff
@@ -125,11 +126,13 @@ async fn write_transaction_file(
     Ok(file_name)
 }
 
+// Return true
 fn check_transaction(
```

Comment:
maybe instead of having a `check_transaction`, now this makes more sense to be a `rebase_transaction`? So that instead of simply checking if 2 transactions have conflict, a `transaction` can be updated based on the information in the `other_transaction` to remove conflict.